src/tcgwars/logic/impl/gen7/SunMoonPromos.groovy

```diff
@@ -2619,9 +2677,102 @@ public enum SunMoonPromos implements LogicCardInfo {
         return copy(UnbrokenBonds.LUCARIO_MELMETAL_GX_120, this);
       case GARCHOMP_GIRATINA_GX_SM193:
         return copy(UnifiedMinds.GARCHOMP_GIRATINA_GX_146, this);
+      case DETECTIVE_PIKACHU_SM194:
+        return basic (this, hp:HP090, type:L, retreatCost:2) {
+        weakness F
+        resistance M, MINUS20
+        move "Brilliant Deduction", {
+          text "Look at the top 4 cards of your deck and put 1 of them into your hand. Shuffle the other cards back into your deck."
+          energyCost L
+          attackRequirement {
+            assert my.deck : "Your deck is empty."
+          }
+          onAttack {
+            def top4 = my.deck.subList(0,4)
+            top4.showToMe("The top 4 cards of your deck.")
+            top4.select(max:1,"Choose a card to put in your hand.").moveTo(my.hand)
+            shuffleDeck()
+            }
+          }
+        };
+      case CHARIZARD_GX_SM195:
+        return evolution (this, from:"Charmeleon", hp:HP250, type:R, retreatCost:4) {
+          weakness W
+          move "Raging Destruction", {
+            text "Discard the top 8 cards of your deck. If any of those cards are [R] Energy cards, attach them to this Pokémon."
+            energyCost R
+            attackRequirement {
+              assert my.deck : "Your deck is empty."
+            }
+            onAttack {
+              def top8 = my.deck.subList(0,8)
+              def list = top8.filterByType(BASIC_ENERGY).filterByEnergyType(R)
+              def num = list.size()
+              top8.showToMe("The top 8 cards of your deck.")
+              list.showToMe("These [R] Energies will be attached.")
+              if (num){
+                list.each{attachEnergy(self, it)}
+                }
+              }
+              my.deck.subList(0,8-num).discard()
+            }
+          }
+          move "Steam Artillery", {
+            text "200 damage."
+            energyCost R, R, C, C, C
+            onAttack {
+              damage 200
+            }
+          }
+          move "Dreadful Flames GX", {
+            text "250 damage. Discard an Energy from each of your opponent's Pokémon. (You can't use more than 1 GX attack in a game.)"
+            energyCost R, R, C, C, C
+            attackRequirement {
+              gxCheck()
+            }
+            onAttack {
+              gxPerform()
+              damage 250
+              opp.all.each{
+                if (it.cards.filterByType(ENERGY)) {
```

Comment:
This if block appears to be missing a closing bracket

Reply:
Correct. Made this adjustment